tic-tac-toe/src/components/Game.js

```diff
@@ -0,0 +1,58 @@
+import React, { useState } from "react";
+import { calculateWinner } from "../helper";
+import Board from "./Board";
+
+const Game = () => {
+    const [history, setHistory] = useState([Array(9).fill(null)]);
+    const [stepNumber, setStepNumber] = useState(0);
+    const [xIsNext, setXisNext] = useState(true);
+    const winner = calculateWinner(history[stepNumber]);
+    const xO = xIsNext ? "x" : "o";
+
+    const handleClick = (i) => {
+        const historyPoint = history.slice(0, stepNumber + 1);
+        const current = historyPoint[stepNumber];
+        const squares = [...current];
+
+        // return if won or occupied
+        if(winner || squares[i]) return;
+
+        // select square
+        squares[i] = xO;
+        setHistory([...historyPoint, squares]);
+        setStepNumber(historyPoint.length);
+        setXisNext(!xIsNext);
+    };
+
+    const jumpTo = (step)=> {
+        setStepNumber(step);
+        setXisNext(step % 2 === 0);
+
+    };
+
+    const renderMoves = () =>
+        history.map((_step, move) => {
```

Comment:
This could benefit from useMemo since we only need to update the moves when history state changes!

The react documentation is actually good for this one: https://reactjs.org/docs/hooks-reference.html#usememo